docs/diagnostics.md

```diff
@@ -36,6 +36,16 @@ From here, you can either simply update the error text, or you can use some of t
 
 If you're including data from user code in an error message, it's important to also write a test that verifies the exact error message for a given string of F# code.
 
+Note that the .NET SDK generally follows the policy that new diagnostics are introduced only alongside a language version increment, and F# now tries to follow the same policy (though historically this has not always happened).
+This is true even if your diagnostic is off by default.
+So if you create a new diagnostic by adding it to [FsComp.txt](https://github.com/dotnet/fsharp/blob/main/src/Compiler/FSComp.txt), you should add a new language feature at the same time.
+Concretely, consider following this procedure:
+
+1. Create a pull request that reserves an error message. (That way, you can avoid other people trampling over your change to FSComp.txt while you're implementing the actual feature.) [Here's a historical example.](https://github.com/dotnet/fsharp/pull/14642)
+1. In the same pull request, reserve the new feature. [Here's a historical example](https://github.com/dotnet/fsharp/pull/15315/) (although it would have been better practice to combine this into the previous pull request, and also this pull request [should not have registered the feature](https://github.com/dotnet/fsharp/pull/15315/files#r1220330247) as belonging to a particular language version).
+1. Now you can go ahead and implement the diagnostic. This is the point at which you would register the language feature in some particular language version (likely the preview language version). It is often best practice to introduce it at a lower level than you ultimately intend (e.g. informational rather than warning, or warning rather than error), to permit a more gradual rollout. See the section on "[Enabling a warning or error by default](#enabling-a-warning-or-error-by-default)" for more information.
```

Comment:
Again - good general guidance but I don't think we really follow it here since, well, people are lazy to really follow this gradual increase of diagnostic panic and we don't have (and probably won't have) processes for this. 

So I think it's a good dev practice (like e.g. properly implemented feature flags) but this might mislead contributors who could take this too seriously.

That said, I am not against having this here, just voicing my opinion and happy to hear other opinions.